caps/eth.md

```diff
@@ -157,6 +203,48 @@ specification.
 
 - `totalDifficulty` is the total difficulty of the block (aka score).
 
+### NewPooledTransactionHashes (0x08)
+
+`[hash_0: B_32, hash_1: B_32, ...]`
+
+This message announces one or more transactions that have appeared in the network and
+which have not yet been included in a block. To be maximally helpful, nodes should inform
+peers of all transactions that they may not be aware of.
+
+The recommended soft limit for this message is 4096 hashes (128 KiB).
+
+Nodes should only announce hashes of transactions that the remote peer could reasonably be
+considered not to know, but it is better to return more transactions than to have a nonce
+gap in the pool.
+
+### GetPooledTransactions (0x09)
+
+`[hash_0: B_32, hash_1: B_32, ...]`
+
+This message requests transactions from the recipient's transaction pool by hash.
+
+The recommended soft limit for GetPooledTransactions requests is 256 hashes (8 KiB). The
+recipient may enforce an arbitrary limit on the response (size or serving time), which
+must not be considered a protocol violation.
+
+### PooledTransactions (0x0a)
+
+`[[nonce: P, receivingAddress: B_20, value: P, ...], ...]`
+
+This is the response to GetPooledTransactions, returning the requested transactions from
+the local pool. The items in the list are transactions in the format described in the main
+Ethereum specification.
+
+The transactions must be in same order as in the request, but it is OK to skip
+transactions which are not available. This way, if the response size limit is reached,
+requesters will know which hashes to request again (everything starting from the last
+returned transaction) and which to assume unavailable (all gaps before the last returned
+transaction).
+
+A peer may respond with an empty list iff none of the hashes match transactions in its
+pool. It is allowed to announce a transaction that will not be served later if it gets
```

Comment:
@karalabe This sentence "...if it gets included in a block in between" is copied from the EIP, and the meaning isn't 100% clear to me. Can you explain what it means?

Reply:
First peer A announces it, then it becomes included in a block. Peer A will clean it out from it's pool, and thus no longer have it accessible in memory. Since nodes are not expected to do disk-accesses to serve announce-responses, the node will not deliver it, despite earlier having announced it.

Reply:
OK. Makes sense. I'll rephrase it to be more clear.